monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,301 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for check_collectd_mlab check."""
+
+import os
+import socket
+import time
+import unittest2 as unittest
+
+# Third-party modules.
+import mock
+
+# Module under test.
+import check_collectd_mlab
+
+# R0904: Too many public methods. Hard to avoid for unit tests.
+# pylint: disable=R0904
+
+
+class MLabNagiosSocketTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+
+  def testunit_sock_sendcmd_RETURNS_successfully(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = list('1 default reply\n')
+
+    returned_value = check_collectd_mlab.sock_sendcmd(
+        mock_sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 1)
+    mock_sock.send.assert_called_with('GETVAL "whatever"\n')
+
+  def testunit_sock_sendcmd_WHEN_receive_bad_reply_RETURNS_zero(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = list('not-a-number junk\n')
+
+    returned_value = check_collectd_mlab.sock_sendcmd(
+        mock_sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 0)
+    mock_sock.send.assert_called_with('GETVAL "whatever"\n')
+
+  def testunit_sock_connect_WHEN_no_socket_RAISES_Error(self):
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionError,
+        check_collectd_mlab.sock_connect, 'no_socket_name')
+
+  def testunit_sock_readline_WHEN_socket_error_RAISES_Error(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = socket.error('fake error')
+
+    self.assertRaises(
+        check_collectd_mlab.SocketReadlineError,
+        check_collectd_mlab.sock_readline, mock_sock)
+
+
+class MLabCollectdAssertionTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+    check_collectd_mlab.COLLECTD_BIN = (
+        os.path.join(self._testdata_dir, 'fake_bin'))
+    check_collectd_mlab.COLLECTD_NAGIOS = (
+        os.path.join(self._testdata_dir, 'fake_bin'))
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = (
+        os.path.join(self._testdata_dir, 'fake_socket'))
+    check_collectd_mlab.VSYSPATH_BACKEND = (
+        os.path.join(self._testdata_dir, 'fake_backend'))
+    check_collectd_mlab.VSYSPATH_SLICE = (
+        os.path.join(self._testdata_dir, 'fake_slice'))
+
+  @mock.patch('check_collectd_mlab.sock_connect')
+  def testunit_assert_collectd_responds_WHEN_sock_sendcmd_fails(
```

Comment:
I'm afraid I don't understand. Is the original method name unclear and/or the test name? The method under test is `assert_collectd_responds()`.

Maybe related, it looks like I use the suffix "_RAISES_error" inconsistently. In some cases this couldn't be added within 80 columns. :-/  I'm inclined to remove "_RAISES_error" suffix for the tests in the `MLabCollectdAssertionTests` class.